introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "plt.xlabel('Date')\n",
+    "plt.title('Chicago Speed Camera Violations')\n",
+    "plt.legend(loc='upper center', bbox_to_anchor=(0.5, -0.05), shadow=False, ncol=4)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We define prediction length as 30 days, and split the data with last 30 days of data as test data. We use rest of the data for training of the model. We can use the last 30 days of data to evaluate the accuracy of our trained model. We write the training and test data files in JSON format in the S3 bucket. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prediction_length = 30\n",
+    "\n",
+    "# Split the data for training and validation/hold out\n",
+    "violation_list_training = []\n",
+    "for i in violation_list:\n",
+    "    violation_list_training.append((i[:-prediction_length]))\n",
+    "\n",
+    "def series_to_obj(ts, cat=None):\n",
+    "    obj = {\"start\": str(ts.index[0]), \"target\": list(ts)}\n",
+    "    if cat:\n",
+    "        obj[\"cat\"] = cat\n",
+    "    return obj\n",
+    "\n",
+    "def series_to_jsonline(ts, cat=None):\n",
+    "    return json.dumps(series_to_obj(ts, cat))\n",
+    "\n",
+    "encoding = \"utf-8\"\n",
+    "s3filesystem = s3fs.S3FileSystem()\n",
+    "\n",
+    "with s3filesystem.open(train_data_path, 'wb') as fp:\n",
+    "    for ts in violation_list_training:\n",
+    "        fp.write(series_to_jsonline(ts).encode(encoding))\n",
+    "        fp.write('\\n'.encode(encoding))\n",
+    "\n",
+    "with s3filesystem.open(test_data_path, 'wb') as fp:\n",
+    "    for ts in violation_list:\n",
+    "        fp.write(series_to_jsonline(ts).encode(encoding))\n",
+    "        fp.write('\\n'.encode(encoding))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## Train\n",
+    "\n",
+    "We use [SageMaker Python SDK](https://sagemaker.readthedocs.io/en/stable/) to create [estimator](https://sagemaker.readthedocs.io/en/stable/estimators.html) object to kick off training job. The train_use_spot parameter indicates the use of [managed spot training](https://docs.aws.amazon.com/sagemaker/latest/dg/model-managed-spot-training.html). The training will run at most 1 hour (3600 seconds). \n",
+    "\n",
+    "We use the [Automatic model tuning] (https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning.html) or Hyperparameter optimization for identifying the best values for the [DeepAR hyperparameters](https://docs.aws.amazon.com/sagemaker/latest/dg/deepar_hyperparameters.html). The Automatic model tuning job will kick of 10 parallel jobs (set by by max_parallel_jobs) to search the best hyperparameters for this dataset. The jobs will try to minimize the root mean square error on the test dataset using predicted and actual values.\n",
```

Comment:
* [formatting] remove the space after the first "]"
* [nitpick] "Automatic model tuning" --> "Automatic Model Tuning"

Reply:
fixed